neqo-transport/src/crypto.rs

```diff
@@ -1512,16 +1513,33 @@ impl CryptoStreams {
             builder.encode_varint(crate::frame::FRAME_TYPE_CRYPTO);
             builder.encode_varint(offset);
             builder.encode_vvec(&data[..length]);
+            Some((offset, length))
+        }
 
-            cs.tx.mark_as_sent(offset, length);
-
-            qdebug!("CRYPTO for {} offset={}, len={}", space, offset, length);
-            tokens.push(RecoveryToken::Crypto(CryptoRecoveryToken {
-                space,
-                offset,
-                length,
-            }));
-            stats.crypto += 1;
+        let cs = self.get_mut(space).unwrap();
+        if let Some((offset, data)) = cs.tx.next_bytes() {
+            let written = if let Some(sni) = find_sni(data) {
```

Comment:
I just realized that maybe you should check whether offset is zero.  Or maybe just comment that this will sometimes split something that isn't an SNI.